### ISSUE TITLE
Response type for code 200 could not be set.

### ISSUE THREAD

**Issue by MiErnst** [springfox/springfox#603]
In version 0.9.5 the response code 200 is handled very differently to other return codes. What's the reason for this?

Nevertheless the method `DefaultResponseMessageReader.applyReturnTypeOverride(SwaggerGlobalSettings, HandlerMethod, Map<Integer, ResponseMessage>)` always overrides the value of the `response` property. No chance to change this currently:
```java
@ApiResponses({ @ApiResponse(code = 200, message = "My Foo", response = MyClass.class)})
public Callable<ResponseEntity<?>> doSoemthing() {
```
will be generated to
```json
{
    "code": 200,
    "message": "My Foo",
    "responseModel": "Callable«ResponseEntity«object»»"
}
```

If the code would be 201, the `responseModel` value would be `MyClass`.

**Comment by dilipkrish** [comment]
@MiErnst this is an existing bug. I believe it should be fixed in the upcoming 2.0 version. Will make a note to check that this is fixed. Thanks for reporting!

**Comment by dilipkrish** [comment]
@MiErnst Closing this issue as its should be fixed in the current snapshot
